modular_skyrat/modules/event_vote/code/event_vote.dm

```diff
@@ -0,0 +1,357 @@
+
+	data["next_vote_time"] = (scheduled - world.time) / 10
+
+	data["show_votes"] = show_votes
+
+	data["previous_events"] = list()
+
+	for(var/datum/round_event_control/iterating_event in previously_run)
+		data["previous_events"] += iterating_event.name
+
+	data["votes"] = list()
+
+	for(var/datum/round_event_control/iterating_event in votes)
+		data["votes"] += list(list(
+			"name" = iterating_event.name,
+			"votes" = LAZYLEN(votes[iterating_event]),
+			"ref" = REF(iterating_event),
+		))
+
+	// Build a list of runnable events.
+	data["event_list"] = list()
+	for(var/datum/round_event_control/iterating_event in SSevents.control)
+		if(!iterating_event.can_spawn_vote(get_active_player_count(TRUE, TRUE, TRUE)))
+			continue
+
+		var/self_selected = istype(iterating_event, check_vote(user.ckey)) ? TRUE : FALSE
+
+		data["event_list"] += list(list(
+			"name" = iterating_event.name,
+			"ref" = REF(iterating_event),
+			"self_vote" = self_selected
+		))
+
+	return data
+
+/datum/controller/subsystem/events/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+
+	if(admin_only && !check_rights(R_ADMIN, FALSE))
+		return
+
+	switch(action)
+		if("register_vote")
+			var/selected_event = locate(params["event_ref"]) in SSevents.control
+			if(!selected_event)
+				return
+			register_vote(usr, selected_event)
+			return
+		if("end_vote")
+			if(!check_rights(R_PERMISSIONS))
+				return
+			end_vote(usr)
+			return
+		if("cancel_vote")
+			if(!check_rights(R_PERMISSIONS))
+				return
+			cancel_vote(usr)
+			return
+		if("start_vote")
+			if(!check_rights(R_PERMISSIONS))
+				return
+			start_vote()
+			return
+		if("start_player_vote")
+			if(!check_rights(R_PERMISSIONS))
+				return
+			var/alert_vote = tgui_alert(usr, "Do you want to show the vote outcome?", "Vote outcome", list("Yes", "No"))
+			if(!alert_vote)
+				return
+			var/public_vote = FALSE
+			if(alert_vote == "Yes")
+				public_vote = TRUE
+			start_player_vote(public_vote)
+			return
+		if("reschedule")
+			if(!check_rights(R_PERMISSIONS))
+				return
+			var/alert = tgui_alert(usr, "Set custom time?", "Custom time", list("Yes", "No"))
+			if(!alert)
+				return
+			var/time
+			if(alert == "Yes")
+				time = tgui_input_number(usr, "Input custom time in seconds", "Custom time", 60, 6000, 1) * 10
+			reschedule_custom(time)
+			message_admins("[key_name_admin(usr)] has rescheduled the event system.")
+			return
+
+// Panel for admins
+/client/proc/event_panel()
+	set category = "Admin.Events"
+	set name = "Event Panel"
+
+	SSevents.ui_interact(usr)
+
+// Panel for everyone
+/mob/verb/event_vote()
+	set category = "OOC"
+	set name = "Event Vote"
+	if(!SSevents.vote_in_progress)
+		to_chat(usr, span_warning("No vote in progress."))
+		return
+	if(SSevents.admin_only && !check_rights(R_ADMIN, FALSE))
+		to_chat(usr, span_warning("You do not have permission to vote."))
+		return
+	SSevents.ui_interact(usr)
```

Comment:
I feel like that should also be a proc that only gets added when a vote is started, so we don't just have it kind of there all the time for no reason, if that makes sense.

Reply:
Probably true, but I'd rather not mess with mob verbs adding/removing tbh, if a runtime occurs etc, it can fuck up, it's only one extra thing in the underused OOC tab tbf